DeprecationWarning: torch.distributed._sharded_tensor

## 🐛 Bug

The latest versions of pytorch-lightning and PyTorch give the following deprecation warning:
```
/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/pytorch_lightning/core/lightning.py:118: in __init__
    self._register_sharded_tensor_state_dict_hooks_if_available()
/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/pytorch_lightning/core/lightning.py:2067: in _register_sharded_tensor_state_dict_hooks_if_available
    from torch.distributed._sharded_tensor import pre_load_state_dict_hook, state_dict_hook
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
    # Keep old package for BC purposes, this file should be removed once
    # everything moves to the `torch.distributed._shard` package.
    import sys
    import torch
    import warnings
    from torch.distributed._shard.sharded_tensor import *  # noqa: F403
>   warnings.warn(
        "torch.distributed._sharded_tensor will be deprecated, use torch.distributed._shard.sharded_tensor instead",
        DeprecationWarning
    )
E   DeprecationWarning: torch.distributed._sharded_tensor will be deprecated, use torch.distributed._shard.sharded_tensor instead
```
This chunk of code has since moved, but `torch.distributed._sharded_tensor` is still being imported in `pytorch_lightning/core/module.py` and `tests/core/test_lightning_module.py`.

### To Reproduce

Run code using `LightningModule` through pytest with the latest version of PyTorch.

### Expected behavior

No DeprecationWarning.

### Environment

- PyTorch Lightning Version (e.g., 1.5.0): 1.6.4
- PyTorch Version (e.g., 1.10): 1.11.0
- Python version (e.g., 3.9): 3.10.4, etc.
- OS (e.g., Linux): Linux, etc.
- CUDA/cuDNN version: N/A
- GPU models and configuration: N/A
- How you installed PyTorch (`conda`, `pip`, source): pip
- If compiling from source, the output of `torch.__config__.show()`: N/A
- Any other relevant information: N/A

--- a/src/pytorch_lightning/core/module.py
+++ b/src/pytorch_lightning/core/module.py
@@ -45,7 +45,7 @@ from pytorch_lightning.utilities.apply_func import apply_to_collection, convert_
 from pytorch_lightning.utilities.cloud_io import get_filesystem
 from pytorch_lightning.utilities.distributed import distributed_available, sync_ddp
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
-from pytorch_lightning.utilities.imports import _TORCH_GREATER_EQUAL_1_12
+from pytorch_lightning.utilities.imports import _TORCH_GREATER_EQUAL_1_11, _TORCH_GREATER_EQUAL_1_12
 from pytorch_lightning.utilities.parsing import collect_init_args
 from pytorch_lightning.utilities.rank_zero import rank_zero_debug, rank_zero_deprecation, rank_zero_warn
 from pytorch_lightning.utilities.signature_utils import is_param_in_hook_signature
@@ -1991,7 +1991,10 @@ class LightningModule(
             rank_zero_debug("Could not register sharded tensor state dict hooks")
             return
 
-        from torch.distributed._sharded_tensor import pre_load_state_dict_hook, state_dict_hook
+        if _TORCH_GREATER_EQUAL_1_11:
+            from torch.distributed._shard.sharded_tensor import pre_load_state_dict_hook, state_dict_hook
+        else:
+            from torch.distributed._sharded_tensor import pre_load_state_dict_hook, state_dict_hook
 
         self._register_state_dict_hook(state_dict_hook)
 

--- a/tests/tests_pytorch/core/test_lightning_module.py
+++ b/tests/tests_pytorch/core/test_lightning_module.py
@@ -299,7 +299,10 @@ def test_device_placement(tmpdir):
 
 @RunIf(min_torch="1.10", skip_windows=True)
 def test_sharded_tensor_state_dict(single_process_pg):
-    from torch.distributed._sharded_tensor import empty as sharded_tensor_empty
+    if _TORCH_GREATER_EQUAL_1_11:
+        from torch.distributed._shard.sharded_tensor import empty as sharded_tensor_empty
+    else:
+        from torch.distributed._sharded_tensor import empty as sharded_tensor_empty
     from torch.distributed._sharding_spec import ChunkShardingSpec
 
     class BoringModelWithShardedTensor(BoringModel):

